{"create_time":"2025-04-05T09:34:50.266Z","id":"90953295-f248-4935-bb7b-a2ba040ca740","source_url":"file:///C:/Users/Dj/.rustup/toolchains/1.30-x86_64-pc-windows-msvc/share/doc/rust/html/std/io/enum.ErrorKind.html","previous_url":"D:\\Lessons\\projects\\rustKG\\rustkg_be\\sandbox\\exams\\effective\\output\\zero-shot-with-rules\\std_io-enum_ErrorKind\\merged_299c4751-4d58-44ce-b63e-384b8da094da.json","relations":["has description","implements trait"],"delete_triples":[{"triples":[["std::io::Cursor","description","A `Cursor` wraps another type and provides it with a `Seek` implementation."],["NotFound","description","An entity was not found, often a file."],["PermissionDenied","description","The operation lacked the necessary privileges to complete."],["ConnectionRefused","description","The connection was refused by the remote server."],["ConnectionReset","description","The connection was reset by the remote server."],["ConnectionAborted","description","The connection was aborted (terminated) by the remote server."],["NotConnected","description","The network operation failed because it was not connected yet."],["AddrInUse","description","A socket address could not be bound because the address is already in use elsewhere."],["AddrNotAvailable","description","A nonexistent interface was requested or the requested address was not local."],["BrokenPipe","description","The operation failed because a pipe was closed."],["AlreadyExists","description","An entity already exists, often a file."],["WouldBlock","description","The operation needs to block to complete, but the blocking operation was requested to not occur."],["InvalidInput","description","A parameter was incorrect."],["InvalidData","description","Data not valid for the operation were encountered. Unlike InvalidInput, this typically means that the operation parameters were valid, however the error was caused by malformed input data. For example, a function that reads a file into a string will error with InvalidData if the file's contents are not valid UTF-8."],["TimedOut","description","The I/O operation's timeout expired, causing it to be canceled."],["WriteZero","description","An error returned when an operation could not be completed because a call to `write` returned `Ok(0)`."],["Interrupted","description","This operation was interrupted. Interrupted operations can typically be retried."],["Other","description","Any I/O error not part of this list."],["UnexpectedEof","description","An error returned when an operation could not be completed because an \"end of file\" was reached prematurely."]],"startIndex":24346,"endIndex":64129},{"triples":[],"startIndex":30329,"endIndex":68137}],"add_triples":[{"triples":[["std::io::Cursor","has description","A `Cursor` wraps an in-memory buffer and provides it with a `Seek` implementation."],["std::io::ErrorKind","implements trait","std::cmp::PartialOrd"],["std::io::ErrorKind","implements trait","std::hash::Hash"],["std::io::ErrorKind","implements trait","std::cmp::Ord"],["std::io::ErrorKind::NotFound","has description","An entity was not found, often a file."],["std::io::ErrorKind::PermissionDenied","has description","The operation lacked the necessary privileges to complete."],["std::io::ErrorKind::ConnectionRefused","has description","The connection was refused by the remote server."],["std::io::ErrorKind::ConnectionReset","has description","The connection was reset by the remote server."],["std::io::ErrorKind::ConnectionAborted","has description","The connection was aborted (terminated) by the remote server."],["std::io::ErrorKind::NotConnected","has description","The network operation failed because it was not connected yet."],["std::io::ErrorKind::AddrInUse","has description","A socket address could not be bound because the address is already in use elsewhere."],["std::io::ErrorKind::AddrNotAvailable","has description","A nonexistent interface was requested or the requested address was not local."],["std::io::ErrorKind::BrokenPipe","has description","The operation failed because a pipe was closed."],["std::io::ErrorKind::AlreadyExists","has description","An entity already exists, often a file."],["std::io::ErrorKind::WouldBlock","has description","The operation needs to block to complete, but the blocking operation was requested to not occur."],["std::io::ErrorKind::InvalidInput","has description","A parameter was incorrect."],["std::io::ErrorKind::InvalidData","has description","Data not valid for the operation were encountered. Unlike InvalidInput, this typically means that the operation parameters were valid, however the error was caused by malformed input data. For example, a function that reads a file into a string will error with InvalidData if the file's contents are not valid UTF-8."],["std::io::ErrorKind::TimedOut","has description","The I/O operation's timeout expired, causing it to be canceled."],["std::io::ErrorKind::WriteZero","has description","An error returned when an operation could not be completed because a call to `write` returned `Ok(0)`."],["std::io::ErrorKind::Interrupted","has description","This operation was interrupted. Interrupted operations can typically be retried."],["std::io::ErrorKind::Other","has description","Any I/O error not part of this list."],["std::io::ErrorKind::UnexpectedEof","has description","An error returned when an operation could not be completed because an \"end of file\" was reached prematurely."]],"startIndex":24346,"endIndex":64129},{"triples":[],"startIndex":30329,"endIndex":68137}],"metrics":{"escapeCount":0,"costTime":544955,"totalUsage":{"prompt_tokens":38648,"completion_tokens":15086,"cached_tokens":0,"reasoning_tokens":13812},"chunkLength":2,"chunksInfo":[{"chunkIndex":0,"strLength":39783,"costTime":278629},{"chunkIndex":1,"strLength":37808,"costTime":266326}]},"model":"deepseek-r1-250120","extraInfo":{},"changes_info_triples":[["NotFound","renamed to","std::io::ErrorKind::NotFound"],["PermissionDenied","renamed to","std::io::ErrorKind::PermissionDenied"],["ConnectionRefused","renamed to","std::io::ErrorKind::ConnectionRefused"],["ConnectionReset","renamed to","std::io::ErrorKind::ConnectionReset"],["ConnectionAborted","renamed to","std::io::ErrorKind::ConnectionAborted"],["NotConnected","renamed to","std::io::ErrorKind::NotConnected"],["AddrInUse","renamed to","std::io::ErrorKind::AddrInUse"],["AddrNotAvailable","renamed to","std::io::ErrorKind::AddrNotAvailable"],["BrokenPipe","renamed to","std::io::ErrorKind::BrokenPipe"],["AlreadyExists","renamed to","std::io::ErrorKind::AlreadyExists"],["WouldBlock","renamed to","std::io::ErrorKind::WouldBlock"],["InvalidInput","renamed to","std::io::ErrorKind::InvalidInput"],["InvalidData","renamed to","std::io::ErrorKind::InvalidData"],["TimedOut","renamed to","std::io::ErrorKind::TimedOut"],["WriteZero","renamed to","std::io::ErrorKind::WriteZero"],["Interrupted","renamed to","std::io::ErrorKind::Interrupted"],["Other","renamed to","std::io::ErrorKind::Other"],["UnexpectedEof","renamed to","std::io::ErrorKind::UnexpectedEof"],["std::io::ErrorKind","adds feature","std::cmp::PartialOrd"],["std::io::ErrorKind","adds feature","std::hash::Hash"],["std::io::ErrorKind","adds feature","std::cmp::Ord"]]}